api/friendlyGet.js

```diff
@@ -0,0 +1,29 @@
+import _ from 'lodash';
+import allSettled from 'promise.allsettled';
+
+class FriendlyGet {
+  constructor() {
+    this.requests = {};
+  }
+
+  with(key, promise) {
+    this.requests[key] = promise;
+
+    return this;
+  }
+
+  async get() {
+    let keys = _.keys(this.requests);
+    let asyncValues = await allSettled(_.values(this.requests));
+
+    asyncValues = _.map(asyncValues, x => {
+      return x.status === 'fulfilled' ? x.value : null;
```

Comment:
Should `with()` support an optional default value to permit the caller to supply something other than `null`?   Just a though, I don't feel strongly at all about this